### PR TITLE
Torch dynamic shapes

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Description

Implements dynamic shape support for the torch backend export, addressing the TODO in keras/src/backend/torch/export.py.

Previously, exporting models with the torch backend would bake in fixed shapes, causing tests with different batch sizes to fail.

## Changes

- Modified track_and_add_endpoint in keras/src/backend/torch/export.py to:
  - Build dynamic_shapes from input_signature where shape has None
  - Use torch.export.Dim to mark dynamic dimensions
  - Changed replace_none_number from 1 to 2 to avoid torch dimension specialization
- Removed early returns in tests that were skipping dynamic shape verification:
  - keras/src/export/saved_model_test.py
  - keras/src/export/onnx_test.py

### CODE REVIEW

File: keras/src/backend/torch/export.py [hunk 1]
```diff
@@ -36,15 +36,30 @@ def track_and_add_endpoint(self, name, resource, input_signature, **kwargs):
...
+        # Build dynamic_shapes from input_signature where shape has None
+        dynamic_shapes = []
+        for spec_idx, spec in enumerate(input_signature):
+            shape = getattr(spec, "shape", None)
+            if shape is not None:
+                dim_spec = {}
+                for dim_idx, dim_val in enumerate(shape):
+                    if dim_val is None:
+                        dim_spec[dim_idx] = torch.export.Dim(
+                            f"dim_{spec_idx}_{dim_idx}", min=1
```
Comment (hertschuh) [inline keras/src/backend/torch/export.py]: I think dimension 0 should be shared between the various inputs and should be named `batch`. Otherwise you may run into issues.

https://github.com/keras-team/keras/blob/master/keras/src/backend/jax/export.py#L158-L159

File: keras/src/backend/torch/export.py [hunk 2]
```diff
@@ -36,15 +36,30 @@ def track_and_add_endpoint(self, name, resource, input_signature, **kwargs):
...
-            lambda x: convert_spec_to_tensor(x, replace_none_number=1),
+            lambda x: convert_spec_to_tensor(x, replace_none_number=2),
             input_signature,
         )
         sample_inputs = tuple(sample_inputs)
 
+        # Build dynamic_shapes from input_signature where shape has None
+        dynamic_shapes = []
+        for spec_idx, spec in enumerate(input_signature):
+            shape = getattr(spec, "shape", None)
```
Comment (hertschuh) [inline keras/src/backend/torch/export.py]: You can assume that `spec.shape` is defined, no need for `getattr`.

File: keras/src/backend/torch/export.py [hunk 3]
```diff
@@ -36,15 +36,30 @@ def track_and_add_endpoint(self, name, resource, input_signature, **kwargs):
...
+            if shape is not None:
```
Comment (hertschuh) [inline keras/src/backend/torch/export.py]: You can assume that `shape` is not `None`, no need for an if

**Review by hertschuh — COMMENTED** [review]
Thanks for fixing this gap!

**Comment by MarcosAsh** [comment]
Hey just did the changes! thanks for the feedback
